miden-lib/src/transaction/mod.rs

```diff
@@ -161,6 +160,50 @@ impl TransactionKernel {
             .expect("Invalid stack input")
     }
 
+    /// Extends the advice inputs with account data and Merkle proofs.
+    pub fn extend_advice_inputs_for_account(
+        advice_inputs: &mut AdviceInputs,
+        account_header: &AccountHeader,
+        account_code: &AccountCode,
+        storage_header: &AccountStorageHeader,
+        merkle_path: &MerklePath,
+    ) -> Result<(), MerkleError> {
+        let account_id = account_header.id();
+        let account_nonce = account_header.nonce();
+        let vault_root = account_header.vault_root();
+        let storage_root = account_header.storage_commitment();
+        let code_root = account_header.code_commitment();
+
+        let foreign_id_root = Digest::from([account_id.into(), ZERO, ZERO, ZERO]);
+        let foreign_id_and_nonce = [account_id.into(), ZERO, ZERO, account_nonce];
+
+        // Extend the advice inputs with the new data
+        advice_inputs.extend_map([
+            // ACCOUNT_ID -> [ID_AND_NONCE, VAULT_ROOT, STORAGE_ROOT, CODE_ROOT]
+            (
+                foreign_id_root,
+                [
+                    &foreign_id_and_nonce,
+                    vault_root.as_elements(),
+                    storage_root.as_elements(),
+                    code_root.as_elements(),
+                ]
+                .concat(),
```

Comment:
Should this be `AccountHeader::as_elements()` as well? Not sure if that makes sense but it does seem more in line with the others since it does encode all the header information as well

Reply:
I agree!